src/utils/list.h

```diff
@@ -56,9 +56,9 @@ static inline int dl_list_empty(struct dl_list *list) {
   return list->next == list;
 }
 
-static inline ssize_t dl_list_len(struct dl_list *list) {
+static inline unsigned int dl_list_len(struct dl_list *list) {
   struct dl_list *item;
-  ssize_t count = 0;
+  int count = 0;
```

Comment:
```suggestion
  unsigned int count = 0;
```

Reply:
Actually, this is on purpose!

It's taken directly from the hostapd source code: https://w1.fi/cgit/hostap/tree/src/utils/list.h?id=0ccb3b6cf25e0c432de092a5298be96accd18c68#n54

I'm not 100% sure why, maybe it's because in C, signed ints overflowing is undefined behaviour, so the compiler can assume that the loop will never run forever?

If it's important, we could have a look at the `hostapd` git history, and see if they explain why, since if they don't mention it, we could send a patch to hostapd.

Reply:
:+1: 

Once #265 is merged, this wouldn't matter anyway, since we'd be using the hostapd code directly then.